sympy/core/relational.py

```diff
@@ -547,14 +551,39 @@ def _eval_rewrite_as_Add(self, *args, **kwargs):
 
     @property
     def binary_symbols(self):
-        return self.as_Predicate().binary_symbols
+        if S.true in self.args or S.false in self.args:
+            if self.lhs.is_Symbol:
+                return {self.lhs}
+            elif self.rhs.is_Symbol:
+                return {self.rhs}
+        return set()
 
     def _eval_simplify(self, **kwargs):
-        rel = self.as_Predicate().simplify(**kwargs)
-        ret = rel.refine()
-        if isinstance(ret, BooleanAtom):
-            return ret
-        return self.func(*rel.arguments)
+        from .add import Add
+        from sympy.core.expr import Expr
+        from sympy.solvers.solveset import linear_coeffs
+        # standard simplify
+        e = super()._eval_simplify(**kwargs)
```

Comment:
I'm not sure I understand the purpose of these changes. Is this migrating code from somewhere else?

Reply:
This reverts the change from #20965. Back then, the idea was to replace `Eq` with `Q.eq`, so I made `Eq.simplify()` call `Q.eq.simplify()` in order to avoid parallel code implementation during the migration process. As discussed in mailing list, `Q.eq` will not replace `Eq` so I revert the change here.

Reply:
This reverts the change from #20965. Back then, the idea was to replace `Eq` with `Q.eq` so I made `Eq.simplify()` call `Q.eq.simplify()` in order to avoid parallel code implementation during the migration process. Since the new design (which was discussed in mailing list) does not make any replacement, I choose to revert that here.

Reply:
Also, please note that every other change in `core/relational` just reverts #20965.